Added evaluation of color operations and literals supporting color alpha

ggetz opened the pull request:
Added literals for `rgba(...)` and `hsla(...)`.

I merged in my changes to `Color.js` into this branch and implemented evaluation of colors.

I do the check for type in the evaluation function, because we cannot know for sure the type of the operands until they are evaluated. In the future we can probably put in a faster evaluation if we know the types before runtime.

I also added tests for color operations and evaluation of literal colors with alpha.

pjcozzi commented:
Merge `AnalyticalGraphicsInc:3d-tiles-style` into your branch.

pjcozzi commented:
> In the future we can probably put in a faster evaluation if we know the types before runtime.

For sure.  When you have specific ideas like this, can you add `PERFORMANCE_IDEA` comments to the code?  We'll likely do the most important ones and then delete the less important ones.

pjcozzi commented on Source/Scene/Expression.js:
```diff
@@ -65,6 +65,16 @@ define([
                 if (defined(val)) {
                     node = new Node(val);
                 }
+            } else if (call === 'rgba') {
```
Just curious, I only looked briefly, but is `rgba` and `hsla` the only ways to define alpha in CSS colors?

Also, I thought alpha goes from 0 to 1 in CSS colors, not 0 to 255.  At least that was the example I saw: http://www.w3schools.com/cssref/css_colors_legal.asp

ggetz replied:
According to that link, those are the only legal css colors.

Yes, you are correct. I will update this.

pjcozzi commented on Source/Scene/Expression.js:
```diff
@@ -127,31 +137,72 @@ define([
+            return Color.multiply(left, right, new Color());
+        } else if (right instanceof Color && typeof(left) === 'number') {
+            return Color.multiplyByScalar(right, left, new Color());
+        } else if (left instanceof Color && typeof(right) === 'number') {
```
When we do syntax errors, which we should start soon, we can check, for example, if `right` was a string here just once when walking the AST.

pjcozzi commented on Specs/Scene/ExpressionSpec.js:
```diff
@@ -51,6 +51,12 @@ defineSuite([
         expression = new Expression(new MockStyleEngine(), 'hsl(0, 0, 1)');
         expect(expression.evaluate(undefined)).toEqual(Color.WHITE);
+
+        expression = new Expression(new MockStyleEngine(), 'rgba(0, 0, 0, 0)');
```
These alpha tests are a bit too trivial.  The `expect` doesn't need to be a `Color` constant, it can be a `Color` you construct (might need to use `Color.equals`).

ggetz commented:
> For sure. When you have specific ideas like this, can you add PERFORMANCE_IDEA comments to the code? We'll likely do the most important ones and then delete the less important ones.

OK I added this.

pjcozzi commented:
Look good, just those comments.

ggetz commented:
I added a step to convert from css `rgba` which takes alpha as a value from 0 to 1 to the cesium color. I also updated the tests and added a less trivial test for colors with alpha channels.

pjcozzi commented:
:+1:
